compiler/rustc_middle/src/mir/interpret/allocation.rs

```diff
@@ -796,8 +796,8 @@ impl<Prov: Provenance, Extra, Bytes: AllocBytes> Allocation<Prov, Extra, Bytes>
         Ok(())
     }
 
-    /// Initialize all previously uninitialized bytes in the entire allocation, and set
-    /// provenance of everything to `Wildcard`. Before calling this, make sure all
+    /// Initialize all previously uninitialized bytes in the entire allocation, but
+    /// do not actually mark them as init. Before calling this, make sure all
     /// provenance in this allocation is exposed!
     pub fn prepare_for_native_access(&mut self) {
```

Comment:
I thought we had agreed that this function should just disappear entirely? I guess I was not clear enough, so here we go:
- please adjust `write_uninit` to fill the range with 0s
- then remove `prepare_for_native_access`, it is not not needed any more.

Reply:
Oh! Sorry, I thought you meant the opposite - I was concerned that having `mark_uninit()` write zeroes by default would be a perf hit. But sure, I'll do that if you think it's an okay tradeoff

Reply:
Ah, that was the misunderstanding then. :)

`write_uninit` already does some things that are probably more expensive than writing a few 0s, so I think it's fine.